metrics.go

```diff
@@ -78,140 +69,110 @@ type PrometheusMetrics struct {
 	registry           prometheus.Registerer
 	totalThreads       prometheus.Counter
 	busyThreads        prometheus.Gauge
-	totalWorkers       map[string]prometheus.Gauge
-	busyWorkers        map[string]prometheus.Gauge
-	readyWorkers       map[string]prometheus.Gauge
-	workerCrashes      map[string]prometheus.Counter
-	workerRestarts     map[string]prometheus.Counter
-	workerRequestTime  map[string]prometheus.Counter
-	workerRequestCount map[string]prometheus.Counter
+	totalWorkers       *prometheus.GaugeVec
+	busyWorkers        *prometheus.GaugeVec
+	readyWorkers       *prometheus.GaugeVec
+	workerCrashes      *prometheus.CounterVec
+	workerRestarts     *prometheus.CounterVec
+	workerRequestTime  *prometheus.CounterVec
+	workerRequestCount *prometheus.CounterVec
 	mu                 sync.Mutex
 }
 
+func initWorkerMetrics(metrics Metrics) {
```

Comment:
This assumes prometheus is always the current metrics implementation. It is for frankenphp (the server) but not necessarily true for frankenphp (the library). There are a number of tools out there that use frankenphp (the library) for other purposes and may use different metrics implementations -- or none at all.

This is why current setup is during declaration of the total workers -- this is the first place metrics knows what workers are available.